tests/test_a2a_client.py

```diff
@@ -339,23 +350,17 @@ async def test_send_message_success(mock_ensure, mock_create_client, mock_uuid):
     }
     assert result == expected
     mock_ensure.assert_called_once()
-
-    # Verify client was called with correct message structure
-    mock_client.send_message.assert_called_once()
-    call_args = mock_client.send_message.call_args[0][0]
-    assert isinstance(call_args, SendMessageRequest)
-    assert call_args.id == "request_id_456"
-    assert call_args.params.message.role == Role.user
-    assert call_args.params.message.message_id == "message_id_123"
+    mock_discover.assert_called_once_with("http://test.com")
+    mock_client_factory.create.assert_called_once_with(mock_agent_card)
 
 
 @pytest.mark.asyncio
-@patch.object(A2AClientToolProvider, "_create_a2a_client")
+@patch.object(A2AClientToolProvider, "_discover_agent_card")
```

Comment:
Should these be turned into fixtures to cut down on repetitiveness going forward?